source/common/filter/config_discovery_impl.cc

```diff
@@ -28,84 +28,42 @@ void validateTypeUrlHelper(const std::string& type_url,
     FilterConfigSubscriptionSharedPtr& subscription,
-    const absl::flat_hash_set<std::string>& require_type_urls,
-
-    Server::Configuration::FactoryContext& factory_context,
-    Envoy::Http::FilterFactoryCb default_config, bool last_filter_in_filter_chain,
+    const absl::flat_hash_set<std::string>& require_type_urls, bool last_filter_in_filter_chain,
     const std::string& filter_chain_type)
     : subscription_(subscription), require_type_urls_(require_type_urls),
-      default_configuration_(default_config ? absl::make_optional(default_config) : absl::nullopt),
-      tls_(factory_context.threadLocal()), init_target_("DynamicFilterConfigProviderImpl",
-                                                        [this]() {
-                                                          subscription_->start();
-                                                          // This init target is used to activate
-                                                          // the subscription but not wait for a
-                                                          // response. It is used whenever a default
-                                                          // config is provided to be used while
-                                                          // waiting for a response.
-                                                          init_target_.ready();
-                                                        }),
+      init_target_("DynamicFilterConfigProviderImpl",
+                   [this]() {
+                     subscription_->start();
+                     // This init target is used to activate the subscription but not wait for a
+                     // response. It is used whenever a default config is provided to be used while
+                     // waiting for a response.
+                     init_target_.ready();
+                   }),
       last_filter_in_filter_chain_(last_filter_in_filter_chain),
       filter_chain_type_(filter_chain_type) {
 
   subscription_->filter_config_providers_.insert(this);
-  tls_.set([](Event::Dispatcher&) { return std::make_shared<ThreadLocalConfig>(); });
 }
 
-DynamicFilterConfigProviderImpl::~DynamicFilterConfigProviderImpl() {
+DynamicFilterConfigProviderImplBase::~DynamicFilterConfigProviderImplBase() {
   subscription_->filter_config_providers_.erase(this);
 }
 
-void DynamicFilterConfigProviderImpl::validateTypeUrl(const std::string& type_url) const {
+void DynamicFilterConfigProviderImplBase::validateTypeUrl(const std::string& type_url) const {
   validateTypeUrlHelper(type_url, require_type_urls_);
 }
 
-const std::string& DynamicFilterConfigProviderImpl::name() { return subscription_->name(); }
-
-absl::optional<Envoy::Http::FilterFactoryCb> DynamicFilterConfigProviderImpl::config() {
-  return tls_->config_;
-}
-
-void DynamicFilterConfigProviderImpl::onConfigUpdate(Envoy::Http::FilterFactoryCb config,
```

Comment:
What's the reason for moving these implementations to the header? Is `DynamicFilterConfigProviderImpl` going to become a template?

Reply:
Sorry yes, that's the plan